torch/_inductor/runtime/benchmarking.py

```diff
@@ -195,11 +198,189 @@ def benchmark_gpu(self: Self, _callable: Callable[[], Any], **kwargs: Any) -> fl
+    if not is_fbcode():
+        return feature_config.oss_default
+    if feature_config.local_version is not None:
+        return (
+            feature_config.local_version
+            >= torch._utils_internal.justknobs_getval_int(
+                f"pytorch/benchmarking:{feature_name.upper()}_VERSION"
+            )
+        )
+    return False
+
+
+def maybe_fallback(
+    fn: Callable[Concatenate[Any, P], T]
+) -> Callable[Concatenate[Any, P], T]:
+    """Wrapper that controls feature fallbacks. It is expected that `fn` is a
+    method of one of `Benchmarker`'s subclasses with a corresponding `feature_name`
+    attribute; typing limitations prevent us from declaring this directly. When
+    `fn` is called, in the form `fn(self, ...)`, we will check that the feature
+    `self.feature_name` is enabled; if the feature `feature_name` is not enabled,
+    we will fallback to the parent class' implementation of `fn`.
+    """
+
+    @wraps(fn)
+    def wrapper(self: Any, *args: P.args, **kwargs: P.kwargs) -> T:
+        if not is_feature_enabled(self.feature_name):
+            fallback_fn = getattr(super(self.__class__, self), fn.__name__)
+            counters["inductor"]["benchmarking." + self.feature_name + ".disabled"] += 1
+            logger.debug(
+                "Feature `%s` is disabled, `benchmarking.%s.%s` will fallback to `benchmarking.%s.%s`.",
+                self.feature_name,
+                self.__class__.__name__,
+                fn.__name__,
+                self.__class__.__base__.__name__,
+                fallback_fn.__name__,
+            )
+            return fallback_fn(*args, **kwargs)
+        return fn(self, *args, **kwargs)
+
+    return wrapper
+
+
+class InductorBenchmarker(TritonBenchmarker):
+    feature_name = "inductor_benchmarker"
+
+    @cached_property
+    def L2_cache_size(self: Self) -> int:
+        """Get the L2 cache size, in bytes, of the current device."""
+        device = torch.cuda.current_device()
+        props = torch.cuda.get_device_properties(device)
+        return props.L2_cache_size
+
+    def get_event_pairs(
+        self: Self, iters: int
+    ) -> List[Tuple[torch.cuda.Event, torch.cuda.Event]]:
+        """Get `iters` pairs of CUDA events."""
+        return [
+            (
+                torch.cuda.Event(enable_timing=True),
+                torch.cuda.Event(enable_timing=True),
+            )
+            for _ in range(iters)
+        ]
+
+    def get_event_pairs_min_timing(
+        self: Self, event_pairs: List[Tuple[torch.cuda.Event, torch.cuda.Event]]
+    ) -> float:
+        """Get the minimum timing, in milliseconds, for a group of CUDA event pairs."""
+        return min(
+            [
+                start_event.elapsed_time(end_event)
+                for start_event, end_event in event_pairs
+            ]
+        )
+
+    @maybe_fallback
+    @maybe_time
+    @count
+    def benchmark_gpu(
+        self: Self,
+        _callable: Callable[[], Any],
```

Comment:
Just for posterity - how were these chosen and how do they differ from triton's function ?